src/gen9_vp9_encoder.c

```diff
@@ -1365,8 +1366,8 @@ gen9_vp9_brc_intra_dist_kernel(VADriverContextP ctx,
     gen8_gpe_setup_interface_data(ctx, gpe_context);
 
     memset(&kernel_walker_param, 0, sizeof(kernel_walker_param));
-    kernel_walker_param.resolution_x = vme_context->downscaled_width_in_mb4x;
-    kernel_walker_param.resolution_y = vme_context->downscaled_height_in_mb4x;
+    kernel_walker_param.resolution_x = vp9_state->downscaled_width_4x_in_mb;
+    kernel_walker_param.resolution_y = vp9_state->downscaled_height_4x_in_mb;
```

Comment:
Could you remove downscaled_width_in_mb4x / downscaled_height_in_mb4x from vme_context? Those fields aren't used any more if applying your patch and it is easy to confuse reader if reserving these fields.